### ISSUE TITLE
Optimize Indexes support

### ISSUE THREAD

**Issue by apoorvedave1** [microsoft/hyperspace#60]
### Describe the issue
Blocked by #29 
Once hyperspace has an incremental indexing support, add support for optimizing indexes. This would pick parts of indexes which are made of large number of small files, and combine/compact them into small number of large files. This would improve the performance of the index since the number of index files would be greatly reduced

### To Reproduce

NA
### Expected behavior

`hyperspace.optimize(indexName)` would update indexes to improve their performance
### Environment 

NA

**Comment by apoorvedave1** [comment]
duplicate of #111
